src/API/Site/Controllers/RestAPI/AuthController.php

```diff
@@ -118,6 +118,7 @@ protected function delete_authorize_callback(): callable {
 		return function ( Request $request ) {
 			try {
 				$this->account_service->reset_wpcom_api_authorization();
```

Comment:
I'm thinking if we should call the `delete_wpcom_api_auth_nonce` inside the `reset_wpcom_api_authorization` and `update_wpcom_api_authorization` just to be sure that if we move or call those in other areas of the codebase we don't forget to clear the nonce.